actions/dump-charm-artifacts/action.yml

```diff
@@ -0,0 +1,52 @@
+name: 'Dump Charm Log Artifacts'
+description: 'Dumps to Artifacts useful Juju, Kubernetes, and Charmcraft log information.  Requires kubectl to be set up with a context, and charmcraft to be installed.'
+runs:
+  using: 'composite'
+  steps:
+    - name: Check out repo
+      uses: actions/checkout@v3
+
+    # Setup
+
+    - name: Create artifact collection directory
+      id: mkdir
+      shell: bash
+      if: always()
```

Comment:
Is the `if: always()` needed on each step? what are we preventing with it?

Reply:
iiuc, if we omit `if: always()` then any failure will block the rest of the commands and the action will exit right away.  But the docs here are sparse so I might be wrong.

On the other side, I don't think there's any cost to adding `if: always()` to these, but I might be wrong about that too

Reply:
I'm the other way - I think we always proceed with everything.  If something goes wrong, the partial logs may still be helpful.  I dont think there's any harm to trying everything, even if some fail?

Reply:
But I am in the minority here, so I can just remove all the `always()` calls if that's preferred.

Reply:
@beliaev-maksim maybe [this](https://github.com/canonical/kubeflow-ci/pull/67/files#r1236998052) comment clarifies it and we may want to use `always()` always :)